File: ios/Reguerta/Reguerta/Data/Profiles/InMemorySharedProfileRepository.swift
import Foundation

actor InMemorySharedProfileRepository: SharedProfileRepository {
    private var profiles: [String: SharedProfile] = [
        "member_admin_001": SharedProfile(
            userId: "member_admin_001",
            familyNames: "Ana, Mario y Leo",
            photoUrl: nil,
            about: "Nos encanta la verdura de temporada y venir a recoger los pedidos en familia.",
            updatedAtMillis: 1_742_800_000_000
        ),
        "member_member_001": SharedProfile(
            userId: "member_member_001",
            familyNames: "Marta y Alba",
            photoUrl: nil,
            about: "Somos nuevas en la comunidad y nos apuntamos para aprender a comer mejor.",
            updatedAtMillis: 1_742_860_000_000
        ),
    ]

    func allSharedProfiles() async -> [SharedProfile] {
        profiles.values.sorted { $0.updatedAtMillis > $1.updatedAtMillis }
    }

    func sharedProfile(userId: String) async -> SharedProfile? {
        profiles[userId]
    }

    func upsert(profile: SharedProfile) async -> SharedProfile {
        profiles[profile.userId] = profile
        return profile
    }

    func deleteSharedProfile(userId: String) async -> Bool {
        profiles.removeValue(forKey: userId) != nil
    }
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/data/profiles/InMemorySharedProfileRepository.kt
package com.reguerta.user.data.profiles

import com.reguerta.user.domain.profiles.SharedProfile
import com.reguerta.user.domain.profiles.SharedProfileRepository
import kotlinx.coroutines.sync.Mutex
import kotlinx.coroutines.sync.withLock

class InMemorySharedProfileRepository : SharedProfileRepository {
    private val mutex = Mutex()
    private val profiles = mutableMapOf(
        "member_admin_001" to SharedProfile(
            userId = "member_admin_001",
            familyNames = "Ana, Mario y Leo",
            photoUrl = null,
            about = "Nos encanta la verdura de temporada y venir a recoger los pedidos en familia.",
            updatedAtMillis = 1_742_800_000_000,
        ),
        "member_member_001" to SharedProfile(
            userId = "member_member_001",
            familyNames = "Marta y Alba",
            photoUrl = null,
            about = "Somos nuevas en la comunidad y nos apuntamos para aprender a comer mejor.",
            updatedAtMillis = 1_742_860_000_000,
        ),
    )

    override suspend fun getAllSharedProfiles(): List<SharedProfile> = mutex.withLock {
        profiles.values.sortedByDescending { it.updatedAtMillis }
    }

    override suspend fun getSharedProfile(userId: String): SharedProfile? = mutex.withLock {
        profiles[userId]
    }

    override suspend fun upsertSharedProfile(profile: SharedProfile): SharedProfile = mutex.withLock {
        profiles[profile.userId] = profile
        profile
    }

    override suspend fun deleteSharedProfile(userId: String): Boolean = mutex.withLock {
        profiles.remove(userId) != null
    }
}
